agenda.php

```diff
@@ -309,6 +316,89 @@ protected function getctag() {
         return NULL;
     }
     
+    //if add is true, then add $usermail to the event, otherwise, remove it.
+    function updateAttendee($url, $usermail, $add, $attendee_CN=NULL) {
+        $raw = file_get_contents_safe('./data/' . $url);
+        $etag = file_get_contents_safe('./data/' . $url . '.etag');
+        
+        $vcal = \Sabre\VObject\Reader::read($raw);
+        
+        if($add) {
+            $already_in = false;
+            if(isset($vcal->VEVENT->ATTENDEE)) {
+                foreach($vcal->VEVENT->ATTENDEE as $attendee) {
+                    if(strpos((string)$attendee, $usermail) >= 0) {
+                        $already_in = true;
+                        break;
+                    }
+                }
+            }
+            if($already_in) {
+                $this->log->info("Try to add a already registered attendee");
+                return;
+            }
+            
+            /*$vcal->VEVENT->add(
+              'ATTENDEE',
+              'mailto:' . $usermail,
+              [
+              'RSVP' => 'TRUE',
+              'CN'   => (is_null($attendee_CN)) ? 'Bénévole' : $attendee_CN, //@TODO
+              ]
+              );*/
+            $vcal->VEVENT->add('ATTENDEE', 'mailto:' . $usermail);
+        } else {
+            $already_out = true;
+            throw new NotImplementedException();
+            if($already_out) {
+                $this->log->info("Try to remove an unregistered email");
+                return;
+            }
+        }
+        
+        $new_etag = $this->updateEvent($url, $etag, $vcal->serialize());
+        
+        $this->log->debug($vcal->serialize());
+        if(is_null($new_etag)) {
+            $this->log->info("The server did not answer a new etag after an event update, need to update the local calendar");
+            $this->updateEvents(array($this->url . '/' . $url));
+        } else {
+            file_put_contents_safe("./data/" . $url, $vcal->serialize());
+            file_put_contents_safe("./data/" . $url . ".etag", $new_etag);
+        }
```

Comment:
IIUC we create this event in the local cache but we then get a local cache which is not consistent with the local ctag. However perhaps it's not an issue because the current query may not care about the current ctag and the next one will see that the local ctag is not the latest one and will update the local state, making it consistent.
That being said, since the next query will update and make consistent the local state, perhaps this code here doesn't need to bother writing those files at all?

As you can see I'm still a bit confused here so I'm interested to know if you have some particular thoughts on this

Reply:
Indeed, the local ctag will differ from the one on the server side at the end of the query.
However, by writing files here, the etag of the updated event will be the same on both sides. 
By doing so, the next query will not make a call to `updateEvents()` (equal etag), and the files will not be rewritten.
I hope I have understood your point correctly.